### PR TITLE
Breaking: Transcript skill editing

### PR DESCRIPTION
**Pull request by priyajeet**
Cleans up left over onInteraction calls.
Adds flow typed for text area autosize.
Updates icons.

Changes the way properties are passed from parent UI Element to Child.
For example:

```jsx
<ContentExplorer
    contentPreviewProps = {{
        hasSidebar: true,
        contentSidebarProps:{
            hasSkills: true,
            hasProperties: true,
            hasActivityFeed: true,
            ...
        }
    }},
    ...
/>
```

```jsx
<ContentPreview
    hasSidebar: true,
    contentSidebarProps = {{
        hasSkills: true,
        hasProperties: true,
        hasActivityFeed: true,
        ...
    }},
    ...
/>
```

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: package.json [hunk 1]
```diff
@@ -204,7 +204,7 @@
   },
   "peerDependencies": {
     "axios": "^0.18.0",
-    "box-react-ui": "^22.17.0",
+    "box-react-ui": "^22.30.0",
```
Comment (priyajeet) [inline package.json]: Fix

File: src/components/ContentSidebar/ContentSidebar.js [hunk 2]
```diff
@@ -639,7 +622,12 @@ class ContentSidebar extends PureComponent<Props, State> {
      * @param {string} id - File id
      * @return {void}
      */
-    onSkillChange = (index: number, removes: Array<SkillCardEntry> = [], adds: Array<SkillCardEntry> = []): void => {
+    onSkillChange = (
```
Comment (priyajeet) [inline src/components/ContentSidebar/ContentSidebar.js]: Add tests for this function

File: src/components/ContentSidebar/Skills/Transcript/Transcript.js [hunk 3]
```diff
@@ -4,67 +4,281 @@
...
+
+    /**
+     * Returns a string to be used as a key for transcript rows.
+     * Created using start time, end time and row index.
+     *
+     * @param {SkillCardEntryTimeSlice[]} timeslices - skill entry time slice
+     * @param {number} index - skill entry index
+     * @return {string} A string key for a row
+     */
+    getTimeSliceKey(timeslices?: SkillCardEntryTimeSlice[], index: number): string {
```
Comment (priyajeet) [inline src/components/ContentSidebar/Skills/Transcript/Transcript.js]: Delete, unused

File: src/components/ContentSidebar/Skills/Transcript/Transcript.js [hunk 4]
```diff
@@ -4,67 +4,281 @@
...
     /**
-     * Handles showing or hiding of transcript modal
+     * Called when transcripts gets new properties
      *
      * @private
      * @return {void}
      */
-    toggleModal = (): void => {
+    componentWillReceiveProps(): void {
+        this.setState({ isEditingIndex: this.state.isEditingIndex, newTranscriptText: '' });
```
Comment (priyajeet) [inline src/components/ContentSidebar/Skills/Transcript/Transcript.js]: should set to -1 or undefined

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #315

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

**Pull request merged**